### PR TITLE
Allow custom config file

### PR DESCRIPTION
**Pull request by maciej-firefly**

### CODE REVIEW

File: src/SAML2/Protocol/Saml20LogoutHandler.cs [hunk 1]
```diff
@@ -74,6 +74,10 @@ protected override void Handle(HttpContext context)
                 {
                     idpEndpoint = RetrieveIDPConfiguration(StateService.Get<string>(IdpLoginSessionKey));
                 }
+                else
+                {
+                    idpEndpoint = RetrieveIDP(context);
```
Comment (benjmanning) [inline src/SAML2/Protocol/Saml20LogoutHandler.cs]: Just a little uncertain as to why this is needed? A couple of things:
 - Couldn't RetrieveIDPConfiguration() be extended to contain the config from the file?
 - Is there a risk that the above function and the TransferClient() call below both do a redirect?

File: src/SAML2/Protocol/Saml20MetadataHandler.cs [hunk 2]
```diff
@@ -57,9 +57,10 @@ public override void ProcessRequest(HttpContext context)
                     throw new ArgumentException(ErrorMessages.MetadataSignQueryParameterInvalid);
                 }
             }
-                        
-            context.Response.ContentType = Saml20Constants.MetadataMimetype;
-            context.Response.AddHeader("Content-Disposition", "attachment; filename=\"metadata.xml\"");
+
+            //context.Response.ContentType = Saml20Constants.MetadataMimetype;
+            context.Response.ContentType = "text/xml";
```
Comment (benjmanning) [inline src/SAML2/Protocol/Saml20MetadataHandler.cs]: What was the problem with Saml20Constants.MetadataMimetype (application/samlmetadata+xml)?

**Comment by benjmanning** [comment]
Other than the above and assuming we are happy with the location of the SAML2.config file, LGTM

File: src/SAML2/Protocol/Saml20LogoutHandler.cs [hunk 3]
```diff
@@ -17,6 +17,8 @@ namespace SAML2.Protocol
     /// </summary>
     public class Saml20LogoutHandler : Saml20AbstractEndpointHandler
     {
+        Actions.Actions _actions;
+
```
Comment (gma-ff) [inline src/SAML2/Protocol/Saml20LogoutHandler.cs]: Maybe worth change the name of `Actions.Actions`?

**Comment by gma-ff** [comment]
a small comment of the namespace/type names and some considerations for testing, other than that 👍

**Pull request merged**